drivers/soundwire/generic_bandwidth_allocation.c

```diff
@@ -353,6 +391,10 @@ static int sdw_compute_bus_params(struct sdw_bus *bus)
 		clk_buf = NULL;
 	}
 
+	m_rt = list_last_entry(&bus->m_rt_list, struct sdw_master_runtime, bus_node);
+	s_rt = list_first_entry(&m_rt->slave_rt_list, struct sdw_slave_runtime, m_rt_node);
```

Comment:
not following what this is supposed to do and why you are not dealing with M and S in symmetrical ways, the last/first isn't clear to me at all.

Reply:
I don't think that's correct @bardliao. When there is a new stream, we need to compute the parameters for all streams and all _rt contexts. Think of it this way: at any bank switch the position of all bits could change. It could very well be that the new stream will have higher priority in the bandwidth allocation than the new one.

IOW, the generic allocation is NOT a purely additive mechanism. At every bank switch, the bus allocation is a blank slate and ALL streams/ports can be re-programmed.

Reply:
Changed to `list_for_each_entry(m_rt, &bus->m_rt_list, bus_node)` to check all m_rt